redis_consumer/consumers/polaris_consumer.py

```diff
@@ -0,0 +1,186 @@
+                plt.savefig(img_path)
+
+                # Save coordiates
+                coords_name = '{}.npy'.format(i)
+                if name:
+                    coords_name = '{}_{}'.format(name, coords_name)
+
+                coords_path = os.path.join(tempdir, subdir, coords_name)
+
+                np.save(coords_path, coords[i])
+
+                outpaths.extend([img_path, coords_path])
+                # outpaths.extend([coords_path])
+
+            # Save each prediction image as zip file
+            zip_file = utils.zip_files(outpaths, tempdir)
+
+            # Upload the zip file to cloud storage bucket
+            cleaned = zip_file.replace(tempdir, '')
+            subdir = os.path.dirname(utils.strip_bucket_path(cleaned))
+            subdir = subdir if subdir else None
+            dest, output_url = self.storage.upload(zip_file, subdir=subdir)
+
+        return dest, output_url
+
+    def _consume(self, redis_hash):
+        start = timeit.default_timer()
+        hvals = self.redis.hgetall(redis_hash)
+
+        if hvals.get('status') in self.finished_statuses:
+            self.logger.warning('Found completed hash `%s` with status %s.',
+                                redis_hash, hvals.get('status'))
+            return hvals.get('status')
+
+        self.logger.debug('Found hash to process `%s` with status `%s`.',
+                          redis_hash, hvals.get('status'))
+
+        self.update_key(redis_hash, {
+            'status': 'started',
+            'identity_started': self.name,
+        })
+
+        # Get model_name and version
+        model_name, model_version = settings.POLARIS_MODEL.split(':')
+
+        _ = timeit.default_timer()
+
+        # Load input image
+        fname = hvals.get('input_file_name')
+        image = self.download_image(fname)
+
+        # squeeze extra dimension that is added by get_image
+        image = np.squeeze(image)
+        if image.ndim == 2:
+            # add in the batch and channel dims
+            image = np.expand_dims(image, axis=[0, -1])
+        elif image.ndim == 3:
+            # check if batch first or last
+            if np.shape(image)[2] < np.shape(image)[1]:
+                image = np.rollaxis(image, 2, 0)
+            # add in the channel dim
+            image = np.expand_dims(image, axis=[-1])
+
+        # Pre-process data before sending to the model
+        self.update_key(redis_hash, {
+            'status': 'pre-processing',
+            'download_time': timeit.default_timer() - _,
+        })
+
+        scale = 1
+
+        # detect dimension order and add to redis
+        dim_order = self.detect_dimension_order(image, model_name, model_version)
+        self.update_key(redis_hash, {
+            'dim_order': ','.join(dim_order)
+        })
+
+        # Validate input image
+        if hvals.get('channels'):
+            channels = [int(c) for c in hvals.get('channels').split(',')]
+        else:
+            channels = None
```

Comment:
`dict.get` returns None by default so consider this as an alternative:
```suggestion
        channels = hvals.get('channels')
        if channels:
            channels = [int(c) for c in hvals.get('channels').split(',')]
```

Reply:
If we get rid of the channel drop down, do we still need any of this code?

Reply:
The other consumers all have this code, do you want to update it for all of them to be consistent?